src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -270,16 +273,49 @@ std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_try_translate_expre
         return std::make_shared<JitExpression>(jit_expression_arguments[0], jit_expression_type,
                                                jit_expression_arguments[1], jit_source.add_temporary_value());
       } else if (jit_expression_arguments.size() == 3) {
-        DebugAssert(jit_expression_type == JitExpressionType::Between, "Only Between supported for 3 arguments");
-        auto lower_bound_check =
-            std::make_shared<JitExpression>(jit_expression_arguments[0], JitExpressionType::GreaterThanEquals,
-                                            jit_expression_arguments[1], jit_source.add_temporary_value());
-        auto upper_bound_check =
-            std::make_shared<JitExpression>(jit_expression_arguments[0], JitExpressionType::LessThanEquals,
-                                            jit_expression_arguments[2], jit_source.add_temporary_value());
-
+        std::shared_ptr<JitExpression> lower_bound_check;
+        std::shared_ptr<JitExpression> upper_bound_check;
+
+        switch (jit_expression_type) {
+          case JitExpressionType::BetweenInclusive:
```

Comment:
similar to the rewrite in the ExpressionEvaluator, this can be done with way less code.